dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorNodeSelectionStrategyChannel.java

```diff
@@ -91,14 +91,22 @@ static PinUntilErrorNodeSelectionStrategyChannel of(
             DialoguePinuntilerrorMetrics metrics,
             Random random,
             String channelName) {
+        // We preserve the 'stableIndex' so that calls can be attributed to one host even across reshuffles
```

Comment:
the "stableIndex" is only stable across shuffles not reloads right?

Reply:
yep that's true.

Reply:
That doesn't help us with the atlas case, does it? Those use several hundred unique clients

Reply:
assuming all DialogueChannels get created with uris in the same input order, then I think this would solve their problem